src/core/definePage.spec.ts

```diff
@@ -254,4 +254,128 @@ export default {
       path: '/custom',
     })
   })
+
+  describe('error handling', () => {
+    it('handles syntax errors gracefully when extracting definePage', async () => {
+      const codeWithSyntaxError = `
+<script setup>
+definePage({
+  name: 'test',,  // syntax error: extra comma
+  path: '/test'
+})
+</script>
+
+<template>
+  <div>hello</div>
+</template>
+      `
+      
+      const result = await definePageTransform({
+        code: codeWithSyntaxError,
+        id: 'src/pages/broken.vue?definePage&vue',
+      })
+
+      // Should return empty object instead of crashing
+      expect(result).toBe('export default {}')
```

Comment:
let's also add checks for warnings. There is an API already in place that you can check in other files like src/core/tree.spec.ts. Apply this to other tests that should warn too